### ISSUE TITLE
Feature request: make RNW accessibility

### ISSUE THREAD

**Issue by kopax-polyconseil** [necolas/react-native-web#2189]
**Is your feature request related to a problem? Please describe.**

We did an audit of our application https://passculture.app and there is now a bunch of work to makes the website accessible. I can share the audit privately with @necolas if required.

**Describe a solution you'd like**

Most of the solution cannot be fixed in this package but many of them could be added using props. I'd like to work with you in order to improve the accessibility toolset included in react native web.

**Describe alternatives you've considered**

Doing custom web integration in every project build with react native web, which is costy.

**Additional context**

Happy new year and thanks again for making to project  !

**Comment by necolas** [comment]
RNW supports all ARIA props. If you've got issues to raise please do so. But you've given no details here

**Comment by kopax-polyconseil** [comment]
@necolas I am still studying the topic, but I'll had example in this thread.


# Form tag 

This is an example of `accessibilityRole` being used with `form` to create a `<form></form>` tag and does not produce the desired results :  

```tsx
import React, { ReactNode, SyntheticEvent } from 'react'
import styled from 'styled-components/native'

type FormProps = {
  children: ReactNode
}

function onSubmit(event: SyntheticEvent) {
  event.preventDefault()
}

export function Form({ children }: FormProps) {
  return (
    <FlexForm
      name="search"
      method="POST"
      action={window.location.href}
      onSubmit={onSubmit}
      accessibilityRole="form">
      {children}
    </FlexForm>
  )
}

const FlexForm = styled.View({
  flex: 1,
})
``` 

I would expect to be able to add form attributes and target my form in web with `document.forms.search`.

**Comment by kopax-polyconseil** [comment]
## TypeScript AccessibilityRole

Currently, `@types/react-native` is using `AccessibilityRole` with the following:

```tsx
export type AccessibilityRole =
    | 'none'
    | 'button'
    | 'togglebutton'
    | 'link'
    | 'search'
    | 'image'
    | 'keyboardkey'
    | 'text'
    | 'adjustable'
    | 'imagebutton'
    | 'header'
    | 'summary'
    | 'alert'
    | 'checkbox'
    | 'combobox'
    | 'menu'
    | 'menubar'
    | 'menuitem'
    | 'progressbar'
    | 'radio'
    | 'radiogroup'
    | 'scrollbar'
    | 'spinbutton'
    | 'switch'
    | 'tab'
    | 'tabbar'
    | 'tablist'
    | 'timer'
    | 'list'
    | 'toolbar';
```

This makes `h1`, `h2`, or `form` non supported value necessary for accessibility in TypeScript.

**Comment by necolas** [comment]
If the issue is about the ts types not including all the features available in RNWeb, that's not an issue with RNWeb accessibility. You'll have to customize the ts types you're using to allow you to use the features already supported by RNWeb. This library supports all of ARIA and the `unstable_createElement` API let's you build any custom elements you want.

**Comment by kopax-polyconseil** [comment]
May I ask what is this for https://github.com/necolas/react-native-web/blob/ae13873c2cb9a932bd4f82bcec4dad96c8eed56c/src/modules/AccessibilityUtil/propsToAccessibilityComponent.js#L3-L16 ?
